evap/staff/importers.py

```diff
@@ -97,28 +98,85 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
         assert not self.errors
         # This is safe because the user's email address is checked before in the importer (see #953)
         responsible_dbobj = UserProfile.objects.get(email=self.responsible_email)
-        course = Course(
-            name_de=self.name_de,
+        if self.existing_course:
+            self.existing_course.degrees.add(*self.degrees)
+        else:
+            course = Course(
+                name_de=self.name_de,
+                name_en=self.name_en,
+                type=self.course_type,
+                semester=semester,
+            )
+            course.save()
+            course.responsibles.set([responsible_dbobj])
+            course.degrees.set(self.degrees)
+            evaluation = Evaluation(
+                vote_start_datetime=vote_start_datetime,
+                vote_end_date=vote_end_date,
+                course=course,
+                wait_for_grade_upload_before_publishing=self.is_graded,
+            )
+            evaluation.save()
+            evaluation.contributions.create(
+                evaluation=evaluation,
+                contributor=responsible_dbobj,
+                role=Contribution.Role.EDITOR,
+                textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+            )
+
+    def check_existing_course(self, semester, importer):
+        name_collision = self.check_name_collision(semester)
+        if name_collision:
+            # Only if there are name collisions, it is worth to check for an existing course
+            if self.find_and_set_existing_course(semester, importer):
+                return True
+            if name_collision == "both":
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {}({}) does already exist in this semester.").format(self.name_en, self.name_de)
+                )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} does already exist in this semester.").format(
+                        self.name_en if name_collision == "name_en" else self.name_de
+                    )
+                )
```

Comment:
Previously, the error messages for a name collision were generated depending on whether the german name already existed, or the english name. Additionally, both of them got generated when both names already existed. So I tried to keep this logic but merge the error messages if both names already existed.

Reply:
Sounds good! In the rare case that the German and English names collide with two different courses, this error could give the impression that it is only one. This change is for the better in most cases though, I like :)

Reply:
@janno42 any cases where users need the explicit information what language they are looking at? I could see someone complaining that the course doesn't exist when they just don't find it because they are looking at the English names instead of the German ones.

Reply:
Yep, is https://github.com/e-valuation/EvaP/pull/1683/commits/8c68ddf8638d0da82430f630f19336bf0cd19be6 doing that as expected?

Reply:
Yes 👍